buildSrc/subprojects/ide/src/main/kotlin/org/gradle/gradlebuild/ide/IdePlugin.kt

```diff
@@ -77,230 +56,120 @@ limitations under the License."""
 open class IdePlugin : Plugin<Project> {
 
     override fun apply(project: Project): Unit = project.run {
-        configureIdeaForRootProject()
+        if (providers.environmentVariable("idea.active").getOrElse("false").toBoolean()) {
+            configureIdeaForRootProject()
+        }
     }
 
     private
     fun Project.configureIdeaForRootProject() {
+        val rootProject = this
         apply(plugin = "org.jetbrains.gradle.plugin.idea-ext")
-        apply(plugin = "idea")
-        tasks.named("idea") {
-            doFirst {
-                throw RuntimeException("To import in IntelliJ, please follow the instructions here: https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md#intellij")
+        with(the<IdeaModel>()) {
+            module {
+                excludeDirs = setOf(intTestHomeDir)
             }
-        }
-
-        val rootProject = this
-        plugins.withType<IdeaPlugin> {
-            with(model) {
-                module {
-                    excludeDirs = excludeDirs + rootExcludeDirs
+            project {
+                settings {
+                    configureCopyright()
+                    configureRunConfigurations(rootProject)
+                    doNotDetectFrameworks("android", "web")
                 }
+            }
+        }
+    }
 
-                project {
-                    jdkName = "11.0"
-                    wildcards.add("?*.gradle")
-                    vcs = "Git"
-
-                    settings {
-                        configureCompilerSettings(rootProject)
-                        configureCopyright()
-                        // TODO The idea-ext plugin does not yet support customizing inspections.
-                        // TODO Delete .idea/inspectionProfiles and uncomment the code below when it does
-                        // configureInspections()
-                        configureRunConfigurations(rootProject)
-                        doNotDetectFrameworks("android", "web")
-                        configureSyncTasks(subprojects)
-                    }
+    private
+    fun ProjectSettings.configureCopyright() {
+        copyright {
+            useDefault = GradleCopyright.profileName
+            profiles {
+                create(GradleCopyright.profileName) {
+                    notice = GradleCopyright.notice
+                    keyword = GradleCopyright.keyword
                 }
             }
         }
-        configureJUnitDefaults()
     }
 
     private
     fun ProjectSettings.configureRunConfigurations(rootProject: Project) {
         runConfigurations {
-            create<Application>("Run Gradle") {
-                mainClass = "org.gradle.debug.GradleRunConfiguration"
```

Comment:
We should add a task to run an embedded Gradle from IDEA. The task should take the tasks and the working directory as an argument I suppose. This way we keep the feature and get rid of the run configuration.

Reply:
I see that this can be useful but I personally never missed that.

As this was broken already for some time (and I therefore not used) can we do adding additional task(s) as a follow up issue. If we need it at all. (I think who needs it should create an issue and describe the use case.)

Reply:
IIRC, @adammurdoch uses that functionality.

Reply:
@jjohannes Can you open an issue to introduce a task for the run configuration?

Reply:
https://github.com/gradle/gradle-private/issues/3097